test/links.c

```diff
@@ -1949,6 +1949,7 @@ test_deprec(hid_t fapl, bool new_format)
     hsize_t    num_objs; /* Number of objects in a group */
     char       filename[1024];
     char       tmpstr[1024];
+    int        len = 0; /* Length of comment */
```

Comment:
len should probably be a size_t

Reply:
H5Gget_comment returns int.  I casted len to size_t for strcmp after checking for error.